cli/src/semgrep/commands/ci.py

```diff
@@ -110,7 +110,7 @@ def fix_head_if_github_action(metadata: GitMeta) -> None:
     logger.info(f"Not on head ref: {metadata.head_branch_hash}; checking that out now.")
     git_check_output(["git", "checkout", metadata.head_branch_hash])
 
-    atexit.register(git_check_output, ["git", "checkout", stashed_rev])
+    atexit.register(git_check_output, ["git", "checkout", stashed_rev], os.getcwd())
```

Comment:
I think this is an edge case that only shows up in testing because of how we invoke semgrep during the tests but by the time the atexit runs the working directory has changed, we can pass the current value to git_check_output to ensure this behaves consistently and reduces some noise in the test env.